File: app/src/BaseComponents/themeManager.py
# BaseComponents/themeManager.py

# 定义深色主题颜色方案
DARK_THEME = {
    # 主要颜色
    "primary": "#BB86FC",          # 主色调 - 紫色
    "secondary": "#03DAC6",        # 次要颜色 - 青色
    "background": "#121212",       # 背景颜色 - 深灰
    "surface": "#1E1E1E",          # 表面颜色 - 稍浅的灰
    "error": "#CF6679",            # 错误颜色 - 红色
    
    # 文本颜色
    "on_primary": "#000000",       # 主色调上的文字 - 黑色
    "on_secondary": "#000000",     # 次要颜色上的文字 - 黑色
    "on_background": "#FFFFFF",    # 背景上的文字 - 白色
    "on_surface": "#FFFFFF",       # 表面颜色上的文字 - 白色
    "on_error": "#000000",         # 错误颜色上的文字 - 黑色
    
    # 文本颜色变体
    "text_primary": "#FFFFFF",     # 主要文字颜色 - 白色
    "text_secondary": "#B3B3B3",   # 次要文字颜色 - 浅灰
    
    # 其他颜色
    "divider": "#333333",          # 分割线颜色 - 深灰
}

# 定义浅色主题颜色方案
LIGHT_THEME = {
    # 主要颜色
    "primary": "#6200EE",          # 主色调 - 紫色
    "secondary": "#03DAC6",        # 次要颜色 - 青色
    "background": "#FFFFFF",       # 背景颜色 - 白色
    "surface": "#F5F5F5",          # 表面颜色 - 浅灰
    "error": "#B00020",            # 错误颜色 - 深红
    
    # 文本颜色
    "on_primary": "#FFFFFF",       # 主色调上的文字 - 白色
    "on_secondary": "#000000",     # 次要颜色上的文字 - 黑色
    "on_background": "#000000",    # 背景上的文字 - 黑色
    "on_surface": "#000000",       # 表面颜色上的文字 - 黑色
    "on_error": "#FFFFFF",         # 错误颜色上的文字 - 白色
    
    # 文本颜色变体
    "text_primary": "#000000",     # 主要文字颜色 - 黑色
    "text_secondary": "#666666",   # 次要文字颜色 - 深灰
    
    # 其他颜色
    "divider": "#E0E0E0",          # 分割线颜色 - 浅灰
}

# 当前主题（默认为深色主题）
CURRENT_THEME = DARK_THEME


def get_theme_colors():
    """
    获取当前主题颜色方案
    
    Returns:
        dict: 当前主题的颜色字典
    """
    return CURRENT_THEME


def set_theme(theme_name):
    """
    设置主题
    
    Args:
        theme_name (str): 主题名称 ('dark' 或 'light')
    """
    global CURRENT_THEME
    if theme_name.lower() == 'dark':
        CURRENT_THEME = DARK_THEME
    elif theme_name.lower() == 'light':
        CURRENT_THEME = LIGHT_THEME
    else:
        raise ValueError("不支持的主题名称。请使用 'dark' 或 'light'。")

File: app/duckov_mod_manager/lib/services/theme_manager.dart
// theme_manager.dart
/// 旧版主题管理服务（在原基础上增加多主题支持）
///
/// 要求：
/// - 保持所有旧 API 不变：
///   - ThemeManager.getThemeColors()
///   - ThemeManager.getThemeColor(String)
///   - ThemeManager.headingTextStyle(...)
///   - ThemeManager.bodyTextStyle(...)
///   - 全局函数 headingText(...) / bodyText(...)
/// - 新增 light / dark / system 模式，默认 system 跟随系统明暗。

import 'package:flutter/material.dart';

/// 新增：主题模式枚举
enum AppThemeMode {
  light,
  dark,
  system,
}

class ThemeManager {
  /// 新增：当前主题模式（默认跟随系统）
  static AppThemeMode _currentMode = AppThemeMode.system;

  /// 可选：对外只读，方便外部知道当前模式
  static AppThemeMode get currentMode => _currentMode;

  /// 新增：设置主题模式（调用方可在设置页/启动处使用）
  static void setThemeMode(AppThemeMode mode) {
    _currentMode = mode;
    for (final l in _listeners.toList()) {
      try {
        l();
      } catch (_) {}
    }
  }

  static final Set<VoidCallback> _listeners = <VoidCallback>{};
  static void addListener(VoidCallback listener) {
    _listeners.add(listener);
  }
  static void removeListener(VoidCallback listener) {
    _listeners.remove(listener);
  }

  /// 内部：根据 _currentMode 和系统亮度得到最终模式
  static AppThemeMode _effectiveMode() {
    if (_currentMode != AppThemeMode.system) {
      return _currentMode;
    }
    final brightness =
        WidgetsBinding.instance.platformDispatcher.platformBrightness;
    return brightness == Brightness.dark
        ? AppThemeMode.dark
        : AppThemeMode.light;
  }

  /// 原来的"唯一主题"抽成亮色配置，保持完全不变
  static const Map<String, Color> _lightColors = {
    'primary': Color(0xFF1976D2),
    'on_primary': Colors.white,
    'secondary': Color(0xFF424242),
    'on_secondary': Colors.white,
    'background': Color(0xFFFAFAFA),
    'on_background': Color(0xFF212121),
    'surface': Colors.white,
    'on_surface': Color(0xFF212121),
    'success': Color(0xFF43A047),
    'error': Color(0xFFD32F2F),
    'on_error': Colors.white,
    'text_primary': Color(0xFF212121),
    'text_secondary': Color(0xFF757575),
    'sidebar_background': Color(0xFFF5F5F5),
    'edit_color': Color(0xFF2D8BE3),
  };

  /// 新增：暗色模式配色
  static const Map<String, Color> _darkColors = {
    'primary': Color(0xFF64B5F6),
    'on_primary': Colors.black,
    'secondary': Color(0xFF9E9E9E),
    'on_secondary': Colors.black,
    'background': Color(0xFF1E1E1E),
    'on_background': Color(0xFFE0E0E0),
    'surface': Color(0xFF2D2D2D),
    'on_surface': Color(0xFFE0E0E0),
    'success': Color(0xFF2E7D32),
    'error': Color(0xFFD32F2F),
    'on_error': Colors.black,
    'text_primary': Color(0xFFE0E0E0),
    'text_secondary': Color(0xFFBDBDBD),
    'sidebar_background': Color(0xFF262626),
    'edit_color': Color(0xFF2D8BE3),
  };

  /// 旧 API：获取主题颜色 Map
  /// 实现改为"按当前模式选择 light/dark"，但签名完全不变
  static Map<String, Color> getThemeColors() {
    switch (_effectiveMode()) {
      case AppThemeMode.light:
        return _lightColors;
      case AppThemeMode.dark:
        return _darkColors;
      case AppThemeMode.system:
        // 实际不会走到这里（_effectiveMode 已转换成 light/dark）
        return _lightColors;
    }
  }

  /// 旧 API：获取单个颜色
  static Color getThemeColor(String colorName) {
    final colors = getThemeColors();
    return colors[colorName] ?? Colors.black;
  }

  /// 旧 API：创建标题文本样式
  static TextStyle headingTextStyle({int level = 1, Color? color}) {
    final colors = getThemeColors();
    final baseColor = color ?? colors['text_primary']!;

    switch (level) {
      case 1:
        return TextStyle(
          fontSize: 32,
          fontWeight: FontWeight.bold,
          color: baseColor,
          fontFamily: 'MiSans',
        );
      case 2:
        return TextStyle(
          fontSize: 24,
          fontWeight: FontWeight.w600,
          color: baseColor,
          fontFamily: 'MiSans',
        );
      case 3:
        return TextStyle(
          fontSize: 20,
          fontWeight: FontWeight.w600,
          color: baseColor,
          fontFamily: 'MiSans',
        );
      case 4:
        return TextStyle(
          fontSize: 16,
          fontWeight: FontWeight.w600,
          color: baseColor,
          fontFamily: 'MiSans',
        );
      default:
        return TextStyle(
          fontSize: 16,
          fontWeight: FontWeight.normal,
          color: baseColor,
          fontFamily: 'MiSans',
        );
    }
  }

  /// 旧 API：创建正文文本样式
  static TextStyle bodyTextStyle({double size = 16, Color? color}) {
    final colors = getThemeColors();
    final baseColor = color ?? colors['text_secondary']!;

    return TextStyle(
      fontSize: size,
      fontWeight: FontWeight.normal,
      color: baseColor,
      fontFamily: 'MiSans',
    );
  }
}

/// 旧 API：便捷函数 - 创建标题文本
Widget headingText(String text, {int level = 1, Color? color}) {
  return Text(
    text,
    style: ThemeManager.headingTextStyle(level: level, color: color),
  );
}

/// 旧 API：便捷函数 - 创建正文文本
Widget bodyText(String text, {double size = 16, Color? color}) {
  return Text(
    text,
    style: ThemeManager.bodyTextStyle(size: size, color: color),
  );
}